source/common/tracing/http_tracer_impl.h

```diff
@@ -98,14 +100,29 @@ class HttpTracerUtility {
                             const Http::HeaderMap& request_headers);
 
   /**
-   * 1) Fill in span tags based on the response headers.
-   * 2) Finish active span.
+   * Fills in the active span that represents the downstream request with the proper headers.
+   * Then finishes the span.
    */
   static void finalizeSpan(Span& span, const Http::HeaderMap* request_headers,
```

Comment:
Perhaps I should rename this to `finalizeDownstreamSpan`?

Reply:
Might be a good idea, to make it clear.

Reply:
Done